harness/determined/estimator/_estimator_trial.py

```diff
@@ -706,24 +738,32 @@ def _init_paths(self) -> None:
 
         # Add suffix so that horovod processes don't overwrite each other.
         suffix = str(0) if not self.hvd_config.use else str(hvd.local_rank())
-        if self.load_path is None:
+
+        if self.env.latest_checkpoint is None:
             self.estimator_dir = pathlib.Path(tempfile.mkdtemp(suffix=suffix))
             logging.debug(f"Estimator directory set to {self.estimator_dir}.")
             return
 
-        for callback in self.train_hooks:
-            if isinstance(callback, estimator.RunHook):
-                callback.on_checkpoint_load(str(self.load_path))
-
-        self.estimator_dir = pathlib.Path(tempfile.mkdtemp(suffix=suffix))
-        if self.estimator_dir.exists():
-            shutil.rmtree(str(self.estimator_dir))
-        logging.debug(f"Copying from {self.load_path} to {self.estimator_dir}.")
-        shutil.copytree(str(self.load_path), str(self.estimator_dir))
+        with self._generic._download_initial_checkpoint(self.env.latest_checkpoint) as load_path:
+            for callback in self.train_hooks:
+                if isinstance(callback, estimator.RunHook):
+                    callback.on_checkpoint_load(str(load_path))
 
-        # Calibrate the CheckpointState metadata file to the new location.
-        estimator._update_checkpoint_path_in_state_file(self.estimator_dir)
-        logging.debug(f"Load path set to {self.estimator_dir}.")
+            self.estimator_dir = pathlib.Path(tempfile.mkdtemp(suffix=suffix))
+            if self.estimator_dir.exists():
+                shutil.rmtree(str(self.estimator_dir))
```

Comment:
Ha.  Out of curiosity, I dug up the source of this issue.  Before  October 2019 (`11ba7b4007` to be precise) the lifetime of this tempdir was longer such that it _wasn't_  guaranteed to be clean when this was called.

I've  renamed/indented/unindented this code a couple times since then, and I totally never saw this.

Reply:
ha, turns out this is necessary, because otherwise the `shutil.copytree()` throws an error.  There's a `dirs_exist_ok=False` argument to make it ok, but that's not present in python 3.6.  Restoring this change.